airstack-modules/modules/airstack/social/social.ts

```diff
@@ -0,0 +1,262 @@
+      airUserExtraData.save();
+      airUserExtras.push(airUserExtraData);
+    }
+    // create air profile extras
+    let airProfileExtras = new Array<AirExtra>();
+    for (let i = 0; i < profileExtras.length; i++) {
+      let extra = profileExtras[i];
+      let extraId = userId.concat("-").concat(extra.name);
+      let airProfileExtraData = createAirExtra(
+        extra.name,
+        extra.value,
+        extraId,
+      );
+      airProfileExtraData.save();
+      airProfileExtras.push(airProfileExtraData);
+    }
+    // get air extra data ids
+    let airUserExtraIds = getAirExtraDataEntityIds(airUserExtras);
+    let airProfileExtraIds = getAirExtraDataEntityIds(airProfileExtras);
+    // create air user
+    let airUser = getOrCreateAirUser(chainId, airBlock, dappUserId, toAddress, airUserExtraIds);
+    airUser.save();
+    // create air profile
+    let airProfile = createAirProfile(airBlock, chainId, airUser.id, profileName, tokenId, tokenAddress, airProfileExtraIds);
+    airProfile.save();
+    // create air user registered transaction
+    let airUserRegisteredTransaction = createAirUserRegisteredTransaction(
+      chainId,
+      airBlock,
+      transactionHash,
+      logOrCallIndex,
+      toAddress,
+      airUser.id,
+      airProfile.id,
+      profileName,
+      fromAddress,
+      toAddress,
+      tokenId,
+      tokenAddress,
+      airUserExtraIds,
+      airProfileExtraIds,
+    );
+    airUserRegisteredTransaction.save();
+  }
+
+  /**
+   * @dev this function does not create any entity, it just returns the air extra data entity ids
+   * @param extras air extra data array
+   * @returns air extra data entity ids
+   */
+  function getAirExtraDataEntityIds(
+    extras: AirExtra[],
+  ): string[] {
+    let entityIds = new Array<string>();
+    for (let i = 0; i < extras.length; i++) {
+      let extra = extras[i];
+      entityIds.push(extra.id);
+    }
+    return entityIds as string[];
+  }
+
+  /**
+   * @dev this function does not save the returned entity
+   * @dev this function gets or creates a AirUser entity
+   * @param chainId chain id
+   * @param block air block entity
+   * @param dappUserId dapp user id
+   * @param address air user address (owner of the dappUserId)
+  */
+  function getOrCreateAirUser(
+    chainId: string,
+    block: AirBlock,
+    dappUserId: string,
+    address: string,
+    extras: string[]
+  ): AirUser {
+    let id = chainId.concat('-').concat(dappUserId);
+    let entity = AirUser.load(id);
+    if (entity == null) {
+      entity = new AirUser(id);
+      let airAccount = getOrCreateAirAccount(chainId, address, block);
+      airAccount.save();
+      entity.dappUserId = dappUserId;
+      entity.address = airAccount.id;
+      entity.createdAt = block.id;
+      if (extras.length > 0) entity.extras = extras;
+      entity.lastUpdatedAt = block.id;
+    }
+    return entity as AirUser;
+  }
+
+  /**
+   * @dev this function does not save the returned entity
```

Comment:
This is a create function so it can save the entity.

Reply:
renamed to `getorCreateAirProfile`